soh/src/overlays/actors/ovl_Bg_Spot06_Objects/z_bg_spot06_objects.c

```diff
@@ -425,6 +442,71 @@ void BgSpot06Objects_Update(Actor* thisx, PlayState* play) {
     if (thisx->params == LHO_WATER_TEMPLE_ENTRANCE_LOCK) {
         CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
     }
+
+    // Bail early for water control system for child or non-rando
+    if (LINK_IS_CHILD || !gSaveContext.n64ddFlag) {
+        return;
+    }
+
+    // Begin setup for Lake Hylia water control system
+    if (actionCounter == 0) {
+        // Object containing floor switch data (and ice block data)
+        Object_Spawn(&play->objectCtx, OBJECT_GAMEPLAY_DANGEON_KEEP);
```

Comment:
The floor switch concept was borrowed from the 3ds rando code. This was part of it. I'm not sure what `Object_Spawn` really does. I've also heard that SoH doesn't have object dependency, so not sure if that's related to this.